crates/nu-path/src/helpers.rs

```diff
@@ -10,6 +10,42 @@ pub fn config_dir() -> Option<PathBuf> {
     dirs_next::config_dir()
 }
 
+pub fn vendor_completions_dirs() -> Vec<PathBuf> {
+    let vendor_completions_fn = |mut path: PathBuf| {
+        path.push("nushell");
+        path.push("vendor_completions.d");
+        path
+    };
+    let mut dirs = Vec::new();
+
+    // default global directory
+    let global_default = if cfg!(not(target_os = "windows")) {
+        PathBuf::from("/usr/share")
+    } else {
+        PathBuf::from("C:\\ProgramData")
+    };
+
+    // global directory
+    let global = std::env::var("NUSHELL_GLOBAL_VENDOR_COMPLETIONS_DIR")
```

Comment:
We should probably limit the size of this env var name. Environment space isn't infinite in Windows.

Reply:
Any concrete suggestions? I agree that it's a bit long, but not long enough to actually make a difference when it comes to limits like that.

Reply:
Github marked this review as outdated, but the conversation itself here is not resolved, the var name is still as it was before.

Reply:
My simplification would be `NU_COMPLETION_DIR` since we have precedence with `NU` vs `NUSHELL`
![image](https://github.com/nushell/nushell/assets/343840/54577d3d-8652-472d-9199-52be56adbb3c)

Reply:
That wouldn't allow for differentiating between the global dir and the user dir. What about `NU_COMPLETIONS_DIR` for the user one, and `NU_VENDOR_COMPLETIONS_DIR` for the one managed by the system package manager?

Reply:
What is the point of having a VENDOR one and one without VENDOR? Why do we care, this is just a place to put completions. We don't have that convention in nushell for any other config files. I'm not sure why we'd start with this?

Reply:
why not a `NU_COMPLETION_DIRS: list<path>`? similar to `NU_LIB_DIRS` for modules

Reply:
I'd be ok with that too @amtoine. Then we just add a couple of default folders and make the default_config.nu match.

Reply:
yup, exactly :yum: 

from my experience, `$env.NU_LIB_DIRS` and `$env.NU_PLUGIN_DIRS` are pretty useful and practical objects and notions :ok_hand:

Reply:
I can see how that might be useful, but that also means that if a distribution changes the place where they place their completions and a user has overridden that var while including that folder, they will end up breaking it. I'd therefore be inclined to keep those separated.

If someone wants to modify this so that additionally to the system one a user one a list instead of a single directory, I'd prefer if that was done by someone else in a follow-up PR, to keep this one simpler.